src/dev/flang/be/jvm/classfile/ClassFileConstants.java

```diff
@@ -475,7 +513,7 @@ public String refDescriptor()
       throw new Error("JavaType.refDescriptor only defined for ref type, not for " + this);
     }
 
-    public Expr load(int index)
+    public Expr load(int index, ClassFile cf)
```

Comment:
The `cf` argument is added here to create cpool entries used by the `VerificationType`.  This early creation of cpool entries does not match my intended design for Expr where cpool entries are created late during calls to `code`. This has the advantage that code that turns out to be unused or duplicated or reused in a different class file still works and does not produce any unused cpool entries. 
It would be great if the VerificationType could delay the creation of the cpool entries such that they are not created by a call to `vti(cf)` but during `VerifictionType.write`. This could be done by passing a lambda to `new VerificationType` or by adding arguments with extra data for all cases needed.